pulpcore/pulpcore/app/viewsets/custom_filters.py

```diff
@@ -1,12 +1,53 @@
 """
 This module contains custom filters that might be used by more than one ViewSet.
 """
+from gettext import gettext as _
+from urllib.parse import urlparse
+from uuid import UUID
+
+from django.urls import resolve, Resolver404
 from django_filters import filters
 
+from rest_framework import serializers
+
 
 class CharInFilter(filters.BaseInFilter, filters.CharFilter):
     """
     Enables the user to filter a field by comma separated strings, allowing them to retrieve more
     than one object in a single query.
     """
     pass
+
+
+class HyperlinkRelatedFilter(filters.Filter):
+    """
+    Enables a user to filter by a foreign key using that FK's href
+    """
+
+    def filter(self, qs, value):
+        """
+        Args:
+            qs (django.db.models.query.QuerySet): The Queryset to filter
+            value (string): href representing an instance of foreign key
+
+        Returns:
+            Queryset filtered by the foreign key instance
+        """
+
+        if not value:
+            raise serializers.ValidationError(detail=_('No value supplied for {name} filter.').
+                                              format(name=self.name))
+
+        try:
+            match = resolve(urlparse(value).path)
+        except Resolver404:
+            raise serializers.ValidationError(detail=_('URI not valid: {u}').format(u=value))
+
+        pk = match.kwargs['pk']
+        try:
+            UUID(pk, version=4)
```

Comment:
I am manually validating UUID here otherwise it gets validated much later on when the query gets executed.

Not sure if this is the best way to validate a UUID though.

Reply:
I think this is the best we can do; another PR might follow up by making this value-url-pk workflow some sort of an utility function (to reuse e.g in the `GenericNamedModelViewSet.get_resource()`) but this is cool too.

Btw I'd say this may be worth unit testing for edge conditions.